lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
> Isn't the strong mode's notion of type compatibility "that will require a runtime check"? (Which would be fine here?)

At runtime, `is` and `as` checks against generics in strong mode require that the reified type of the object matches the specified generic type. So `[] as List<String>` will work in dart2js but fail in DDC.

> I guess I don't understand the problem. Is this documented somewhere so I can read up on it?

https://github.com/dart-lang/dev_compiler/blob/master/doc/RUNTIME_SAFETY.md